tflint/runner.go

```diff
@@ -316,12 +317,24 @@ func (r *Runner) EvalExpr(expr hcl.Expression, ret interface{}) (cty.Value, erro
 // EvaluateExpr evaluates the expression and reflects the result in the value of `ret`.
 // In the future, it will be no longer needed because all evaluation requests are invoked from RPC client
 func (r *Runner) EvaluateExpr(expr hcl.Expression, ret interface{}) error {
-	val, err := r.EvalExpr(expr, ret)
+	val, err := r.EvalExpr(expr, ret, cty.Type{})
 	if err != nil {
 		return err
 	}
+	return r.fromCtyValue(val, expr, ret)
+}
 
-	err = gocty.FromCtyValue(val, ret)
+// EvaluateExprType is like EvaluateExpr, but also accepts a known cty.Type to pass to EvalExpr
+func (r *Runner) EvaluateExprType(expr hcl.Expression, ret interface{}, wantType cty.Type) error {
```

Comment:
👍

It will be great if the two APIs are integrated in the future.

Reply:
Agreed. I didn't love this solution, but fixing it seemed like a lot of refactoring which didn't seem prudent.